### ISSUE TITLE
Rails::Delegate should take into account that the receiver is not a class/module constant

### ISSUE THREAD

**Issue by Thermatix** [bbatsov/rubocop#5813]
## Expected behavior

given the following code:

```ruby
class A
  def some_method
    method_resolving_to_a_class.some_method
  end
end
```

I expect rubocop to realise that the receiver `method_resolving_to_a_class`  won't have a value or exist at the class level, only at the instance level and thus accept the above as being valid.

## Actual behavior

It wants me to replace the above with:

```ruby
  delegate :some_method, to: method_resolving_to_a_class
```

which won't work as `method_resolving_to_a_class` doesn't exist at the class level

## Steps to reproduce the problem

No real steps, the code above should cause the cop to throw up a warning

## RuboCop version

Output of `rubocop -V`:

```shell
$ rubocop -V
0.54.0
```

**Comment by mikegee** [comment]
`rubocop --auto-correct` changes your example to:

```ruby
class A
  delegate :some_method, to: :method_resolving_to_a_class
end
```

which will call `method_resolving_to_a_class` on an instance of `A`.

**Comment by Drenmi** [comment]
Unless I'm missing something, something was missed. Feel free to re-open if I'm wrong. 🙂

**Comment by Thermatix** [comment]
The problem is that when call `delegate` it expects `method_resolving_to_a_class` to exist but it doesn't because delegate is called in the context of the class even if the delegation only happens on an instance` as a result when you try to call `delegate` it will crash because `method_resolving_to_a_class` results in a `NoMethodError` because it doesn't exist when you try to delegate, it will only exist at the instance level, not the class level where delegate is called.

**Comment by Thermatix** [comment]
No, the problem is that the Cop shows up at all. Rubocop should realise that `method_resolving_to_a_class` is a method call, not a class constant that therefore may not exist in the context of code run in the class. Therefore doing `delegate :some_method, to: method_resolving_to_a_class` won't work. There's no ambiguity there, `method_resolving_to_a_class` only exists at the instance level, when you try to use it in `delegate` ruby will try to call`method_resolving_to_a_class` to resolve it's value but it won't work as it doesn't exist at the class level.

**Comment by mikegee** [comment]
Rubocop isn't telling you to use
```ruby
delegate :some_method, to: method_resolving_to_a_class
```
it wants
```ruby
delegate :some_method, to: :method_resolving_to_a_class
